tests/unittests/functionality/test_google_oauth.py

```diff
@@ -21,9 +21,9 @@ def test_user_already_logged_in(self):
             register(self.app, 'email@gmail.com', 'test')
             logout(self.app)
             login(self.app, 'email@gmail.com', 'test')
-            self.assertTrue('Open Event' in self.app.get('/gCallback/?state=dummy_state&code=dummy_code',
+            self.assertTrue('Open Event' in self.app.get('/gCallback/?state=dummy_state&code=dummy_code&test=1',
```

Comment:
Yes that's where I'm looking for suggestion. Actually earlier the concept was 
```
 if login.current_user is not None and login.current_user.is_authenticated:
        return redirect(url_for('admin.index'))
```
but since we want to update the user details even if the user is already logged in therefore I had to move this to line 59 in util_routes.py

Originally test expect to get redirected to currect url if is already logged in and sign in from google is pressed. So, I think there are two ways to solve this. First is we can modify the test and use it to know it is actually going to save or saving the data but then we are sending the dummy_code which eventually will not give any client records from google. So may be we can just look for exception throw and test it using ```assertRaises```

Another option I got is to keep the redirect condition in 'gCallback' but will be true only if this is the test case. I feel this is not the right way therefore need your suggestion on this.